scripts/build-motion-library-v5.py

```
@@ -185,6 +185,63 @@ def vnorm(v):
+def _quat_fwd(q):
+    """World +Z basis rotated by quat q=(x,y,z,w) → world forward vector."""
+    x, y, z, w = q
+    return (2 * (x * z + w * y), 2 * (y * z - w * x), 1 - 2 * (x * x + y * y))
+
+
+def _mean_hip_yaw_deg(quats):
+    """|circular mean| yaw (deg off +Z) of the hip's world forward across the
+    clip. ~0 = faces straight ahead; large = the stride is baked sideways
+    (the stylized/quadruped-rig locomotion failure). Uses a CIRCULAR mean
+    (atan2 of summed sin/cos) — a plain arithmetic mean of signed angles
+    cancels across the ±180° wrap (a backward walk jittering +179°/−179°
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)